merkledag/cidversion.go

```diff
@@ -0,0 +1,26 @@
+package merkledag
+
+import (
+	"fmt"
+)
+
+// CidVersion is a a cid version number that is guaranteed to be valid
+type CidVersion struct {
```

Comment:
Also there isn't a reason it couldn't be just a constant?

```
type CidVerision int

const (
CID0 CidVersion = 0
CID1 CidVersion = 1
)
```

You can still have methods on it.